Rename services (#67)
epinio/docs: src/references/settings.md

# Epinio, CLI settings

The Epinio command line client uses a settings file to store
information which has to persist across invocations. This document
discusses various aspects around this.

## Table Of Contents

  - [Location](#location)
  - [Contents](#contents)
  - [Commands](#commands)

## Location

Epinio's settings files is located by default at `~/.config/epinio/settings.yaml`.

The location can be changed from command to command by specifying a
different path with the global command line option `--settings-file`.

> A more permanent change is possible by setting the environment
> variable `EPINIO_SETTINGS` to the desired path.

Reviewer: ```suggestion
The environment variable `EPINIO_SETTINGS` also allows you to set
a different path for the Settings file.
```

## Contents

> Epinio's settings contains

Reviewer: ```suggestion
Epinio's settings contains:
```

>   - The name of the namespace currently targeted.

Reviewer: ```suggestion
  - The name of the `namespace` currently targeted
```

  - Epinio API user name
  - Epinio API password

>   - Epinio API urls (regular and websocket)

Reviewer: ```suggestion
  - Epinio API URLs (regular and websocket)
```

  - Epinio API certificate

The namespace can be changed by running `epinio target` with the
name of the desired namespace as its single argument.

> User name and password are used by the client to authenticate itself
> when talking to Epinio's API server. The `epinio settings update` command
> saves the initial information to the settings.

Reviewer: ```suggestion
User name and password are used by the client for the authentication
against the Epinio's API server. The `epinio settings update` command
saves the initial information to the settings.
```

Reviewer: the "saves the initial information to the settings" means the `settings` are "what" in this context? the in-memory or file settings?

> The installation uses a the wildcard domain `omg.howdoi.website` and the
> `epinio-ca` issuer by default.

Reviewer: ```suggestion
The installation uses a wildcard domain (`omg.howdoi.website`) and the
`epinio-ca` issuer by default.
```

`epinio settings update` saves the associated CA
certificate to the settings so that future invocations of the
client are able to verify the actual certificate when talking to

> Epinio's API server.

Reviewer: same question as above: what is `settings` in this context?

## Commands

> The Epinio command line client currently provides 3 commands
> explicitly targeting the settings. These are:

Reviewer: ```suggestion
The Epinio command line client currently provides three commands
explicitly targeting the settings:
```

  1. `epinio target`

>      As noted in the previous section, this command changes
>      the namespace to use with all other commands.

Reviewer: ```suggestion
     This command sets the `namespace` to be used when running the other commands.
```

  2. `epinio settings show`

>      This command shows the details of the currently stored
>      settings settings. An exception is made for the certificate
>      field, due to its expected size. The command's output only notes
>      if certificate data is present or not.

Reviewer: ```suggestion
     This command shows the details of the current settings. 
     An exception is made for the certificate field, due to its size, 
     and only outputs if certificate data is present or not.
```

  3. `epinio settings update`

>      Epinio allows users to switch between multiple installations (on
>      different clusters) by simply re-targeting the cluster to talk to
>      via changes to the  environment variable `KUBECONFIG`.

Reviewer: ```suggestion
     Epinio allows you to switch between multiple installations (on
     different clusters) by updating the environment variable `KUBECONFIG`
     to the cluster you need to connect.
```

>      When such is done the credentials, cert data, and urls stored in
>      the settings will not match the newly targeted cluster,
>      except by coincidence.
> 
>      To be actually able to talk to the newly targeted installation it
>      is necessary to run this command to refresh the stored
>      credentials, cert data, and urls with information retrieved from
>      the new cluster.

Reviewer: ```suggestion
     Once the `KUBECONFIG` environment variable is updated, you'll
     need to run this command to refresh the stored credentials,
     certificate data and URLs in order to load the settings for the
     selected cluster.
     
     If you don't run this command, the connectivity to the selected
     cluster will fail, as Epinio will still be using the settings for the
     previous cluster.
```
